resources-service/src/resource/resource-model.ts

```diff
@@ -34,15 +34,39 @@ export type SearchParameters = {
 const EMPTY_RESULT = { totalCount: 0, results: [] };
 const MAX_SEARCH_RESULTS = 200;
 
-export const getResource = (
+const groupAttributesByKeys = (
+  attributesWithKeys: (ReferrableResourceAttribute & { key: string })[],
+): Record<string, ReferrableResourceAttribute[]> => {
+  const groupedAttributes: Record<string, ReferrableResourceAttribute[]> = {};
+  attributesWithKeys.forEach(attribute => {
+    groupedAttributes[attribute.key] = groupedAttributes[attribute.key] || [];
+    const { key, ...withoutKey } = attribute;
+    groupedAttributes[attribute.key].push(withoutKey);
+  });
+  return groupedAttributes;
+};
```

Comment:
By reading this function, an attribute of a given `key` can have multiple values. I can see we need this to support multiple languages. Just out of curiosity, is there any other expected use case for reusing the same key?

Reply:
Just for supporting multiple values, so for example, if a resource has multiple 'sites', you can just add a second site value to the key, without having to use a JSONB array, or create a set of special 'list' type tables.

We may want to enforce single values in the data layer at some point, but I thought it was sufficient to enforce that in the UI layer for now to keep the data schema simple